prov/shm/src/smr_progress.c

```diff
@@ -447,23 +457,24 @@ static struct smr_sar_entry *smr_progress_sar(struct smr_cmd *cmd,
 	memcpy(sar_iov, iov, sizeof(*iov) * iov_count);
 	(void) ofi_truncate_iov(sar_iov, &iov_count, cmd->msg.hdr.size);
 
+	sar_entry = ofi_freestack_pop(ep->sar_fs);
+
 	if (cmd->msg.hdr.op == ofi_op_read_req)
-		smr_try_progress_to_sar(peer_smr,
-				smr_sar_pool(ep->region), resp,
-				cmd, iface, device, sar_iov,
-				iov_count, total_len, &next);
+		smr_try_progress_to_sar(ep, peer_smr,
+					smr_sar_pool(ep->region), resp,
+					cmd, iface, device, sar_iov,
+					iov_count, total_len, &next, sar_entry);
 	else
-		smr_try_progress_from_sar(
-				peer_smr, smr_sar_pool(ep->region), resp, cmd,
-				iface, device, sar_iov, iov_count, total_len,
-				&next);
+		smr_try_progress_from_sar(ep,
+		    peer_smr, smr_sar_pool(ep->region), resp, cmd,
```

Comment:
Not sure if this is just my editor but my preference when not aligning to the parantheses for new lines is to align with a tab, not adding spaces